File: leetcode/135.candy/go/candy.go
package candy

func candy(ratings []int) int {
	candyArr := make([]int, len(ratings))
	
	for i := range candyArr {
		candyArr[i] = 1
	}

	for i := 1; i < len(ratings); i++ {
		if ratings[i] > ratings[i-1] {
			candyArr[i] = candyArr[i-1] + 1
		}
	}

	for i := len(ratings) - 2; i >= 0; i-- {
		if ratings[i] > ratings[i+1] && candyArr[i+1]+1 > candyArr[i] {
			candyArr[i] = candyArr[i+1] + 1
		}
	}

	res := 0

	for _, v := range candyArr {
		res += v
	}

	return res
}

File: leetcode/135.candy/ts/candy.ts
const candy = (ratings: number[]): number => {
    const candyArr: number[] = Array(ratings.length).fill(1)

    for (let i = 1; i < ratings.length; i++)
        if (ratings[i] > ratings[i - 1]) candyArr[i] = candyArr[i - 1] + 1

    for (let i = ratings.length - 2; i >= 0; i--)
        if (ratings[i] > ratings[i + 1] && candyArr[i + 1] + 1 > candyArr[i]) candyArr[i] = candyArr[i + 1] + 1

    return candyArr.reduce((acc, v) => acc + v, 0)
}
